src/black/__init__.py

```diff
@@ -474,7 +474,9 @@ def main(
                 out(f"Using configuration in '{config}'.", fg="blue")
 
     error_msg = "Oh no! 💥 💔 💥"
-    if required_version and required_version != __version__:
+    if required_version and (
+        required_version != __version__ or required_version != __version__.split(".")[0]
```

Comment:
This is going to always fail when a required version is specified:
```
if True and (
    "22" != "22.01"  # True
    or "22" != "22"  # False
):
```
```
if True and (
    "22.01" != "22"        # False
    or "22.01" != "22.01"  # True
):
```

Reply:
Lol right.